### DISCUSSION TITLE
Have logging output separate from CLI output.

### DISCUSSION THREAD

**Discussion by bagel99** [openthread/openthread#7966] [Q&A] [answered]
I run tests using the "expect" tool to exercise the CLI.  I would like to get logging output on a separate file descriptor from the CLI output so it can be put in a file and not confuse expect.  Is this possible now, if so what source file do I tweak?  If not, this would be a desirable enhancement.  (This is running on Linux.)

**Comment by abtink** [comment, marked as answer]
@bagel99 you can use the [`OPENTHREAD_CONFIG_LOG_OUTPUT`](https://github.com/openthread/openthread/blob/600ae0999450f3fadcf1134a2f87972de37f48c2/src/core/config/logging.h) to specify where the log output goes to.

- `OPENTHREAD_CONFIG_LOG_OUTPUT_APP` will send the logs to app (CLI or NCP) - which I guess is what you have in your setup.
- `OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED` will use `otPlatLog()` which can be implemented by the platform layer.
   - OT posix platform implementation emits the log to syslog ([posix/logging.cpp](https://github.com/openthread/openthread/blob/600ae0999450f3fadcf1134a2f87972de37f48c2/src/posix/platform/logging.cpp))

**Comment by bagel99** [comment]
So, if I understand correctly, my solution involves creating an alternate otPlatLog() in posix/platform/loggin.cpp that doesn't call vsyslog() but, for example, just calls vprintf(stderr, ...)?

**Reply by abtink** [reply]
Yes. This can work.
